database/migrations/sql/20180604_06_create_users.sql

```diff
@@ -6,27 +6,28 @@ CREATE TABLE user_account (
   user_account_id                  SERIAL NOT NULL UNIQUE,
   user_name                        VARCHAR(100) NOT NULL,
   user_secret                      VARCHAR(64) NOT NULL,
+  qr_code                          VARCHAR,
   gender                           ENUM_gender NOT NULL DEFAULT 'prefer not to say',
-  email                            VARCHAR(100),
+  email                            VARCHAR(100) NOT NULL UNIQUE,
```

Comment:
If we're allowing users to sign up to multiple orgs and merge their accounts at a later point, this field doesn't want to be `UNIQUE`

https://github.com/TwinePlatform/hq/issues/11

Reply:
Do'h

Reply:
Actually, on second thoughts, maybe this can stay.

The Twine-Admin app enforces unique user emails, and in theory has a many-to-one relationship between users and organisations (although in practice I think this collapses to one-to-one).

What I meant to say in TwinePlatform/hq#11 is that we should enforce a 1-to-1 relationship in application code until we have time to support many-to-many.

All this means is an application-level check that the `user_account_id` column in the `user_account_access_role` table is unique.

If users want to sign up for another organisation, I don't see a problem with using a different e-mail. In fact that strikes me as quite natural.

The account merging process can be designed around this.

Reply:
If users are using a different email address to sign up for another account, it will be difficult to merge their accounts automatically. This is the only piece of info that we collect that is unique.

Reply:
> it will be difficult to merge their accounts automatically.

I don't think we need to do it automatically. And in fact, we may not _want_ to do it automatically. It should certainly be something the user opts in to, and they can do a little work to help us do the merge (e.g. enter all e-mail addresses they use, then we can send "are you sure" confirmation e-mails to all those addresses, merge all those that have been confirmed. Or we can design a number of alternatives).

Reply:
Sold! 

Is this PR ready from your side?

Reply:
I think so